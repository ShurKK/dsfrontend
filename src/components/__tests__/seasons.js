import SeasonsComponent from "../seasons"
import React from "react"
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


Enzyme.configure({ adapter: new Adapter() });

describe('<SeasonsComponent />', () => {
  let wrapper;
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState')
  useStateSpy.mockImplementation((init) => [init, setState]);
  const data = {
    seasons: [{name: 'first01', number: 1}, {name: 'second02', number: 2}]
  }

  beforeEach(() => {
    wrapper = Enzyme.shallow(<SeasonsComponent seasons = { data.seasons } />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });


it('renders correctly', () => {  
  //expect(wrapper.exists('.season-picker')).toEqual(true);
  console.log({wrapper});
  expect(wrapper).toMatchSnapshot();
});
  
})
