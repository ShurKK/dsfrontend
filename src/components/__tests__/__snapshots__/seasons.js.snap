// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeasonsComponent /> renders correctly 1`] = `
<Fragment>
  <div
    className="season-picker"
  >
    <SeasonCard
      key="first01"
      season={
        Object {
          "name": "first01",
          "number": 1,
        }
      }
    />
    <SeasonCard
      key="second02"
      season={
        Object {
          "name": "second02",
          "number": 2,
        }
      }
    />
  </div>
</Fragment>
`;
